Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.cpp

```diff
@@ -156,38 +160,104 @@ KeyboardMouse::KeyboardMouse(const LPDIRECTINPUTDEVICE8 kb_device,
+    MONITORINFO current_monitor_info{};
+    current_monitor_info.cbSize = sizeof(MONITORINFO);
+    GetMonitorInfoA(current_monitor, &current_monitor_info);
+    double current_monitor_width{}, current_monitor_height{}, current_monitor_screen_ratio{};
+    current_monitor_width =
+        current_monitor_info.rcMonitor.right - current_monitor_info.rcMonitor.left;
+    current_monitor_height =
+        current_monitor_info.rcMonitor.bottom - current_monitor_info.rcMonitor.top;
+    current_monitor_screen_ratio = current_monitor_width / current_monitor_height;
+
+    // If this doesn't make a ton of sense, I have a shortened and more readable version at the end
+    // of this function
+    m_current_state.cursor.x =
+        ((ControlState)((raw_mouse_point.x) / (ControlState)current_monitor_width) - 0.5) *
+        (Config::Get(Config::MOUSE_SENSITIVITY) *
+         Config::scaling_factor_to_make_mouse_sensitivity_one_for_fullscreen) *
+        current_monitor_screen_ratio;
+    m_current_state.cursor.y =
+        ((ControlState)((raw_mouse_point.y) / (ControlState)current_monitor_height) - 0.5) *
+        (Config::Get(Config::MOUSE_SENSITIVITY) *
+         Config::scaling_factor_to_make_mouse_sensitivity_one_for_fullscreen);
 
-  // Width and height are the size of the rendering window. They could be 0
-  const auto win_width = std::max(rect.right - rect.left, 1l);
-  const auto win_height = std::max(rect.bottom - rect.top, 1l);
+    return;
+  }
 
-  const auto window_scale = g_controller_interface.GetWindowInputScale();
+  // Sets the cursor to the center of the screen during loading to avoid
+  // drift(when booting straight to game). This code will result in drift if boot to pause is used
+  // with the OctagonalMouseJail.
+  if ((OctagonalMouseJail::GetInstance().m_player_requested_center && Host_RendererHasFocus()) ||
+      (::Core::GetState() == ::Core::State::Starting && Host_RendererHasFocus() &&
+       Config::Get(Config::OCTAGONAL_MOUSE_JAIL_ENABLED)))
+  {
+    SetCursorPos(static_cast<int>(center_of_render_window.first),
+                 static_cast<int>(center_of_render_window.second));
+    OctagonalMouseJail::GetInstance().m_player_requested_center = false;
+    return;
+  }
 
-  // Convert the cursor position to a range from -1 to 1.
-  m_state_in.cursor.x = (ControlState(point.x) / win_width * 2 - 1) * window_scale.x;
-  m_state_in.cursor.y = (ControlState(point.y) / win_height * 2 - 1) * window_scale.y;
+  // Everything more than assigning point.x and point.y directly to the
+  // controller's state is to normalize the coordinates since it seems like dolphin
+  // wants the inputs from -1.0 to 1.0
+  //
+  // cursor.x = (mouse.x / screen_width - 0.5) *
+  //            (cursor_sensitivity * screen_ratio);
+  // cursor.y = (mouse.y / screen_height - 0.5) *
+  //            cursor_sensitivity;
+  // ^^^I hope those help with parsing this^^
+  //(cursor sensitivity is also scaled here)
```

Comment:
If you want the formatter off for a few lines, you can do

~~~
// clang-format off
/* code that should not be autoformatted */
// clang-format on
~~~

Reply:
:o ty!